### PR TITLE
Cleanup services

### PR DESCRIPTION
**Pull request by yishaigalatzer**
1. Make explicit registration for nested provider manager (solves a problem with ninject for default MV #953)
2. Remove registation for async nested provider manager
3. Make Action selector a singleton (should assist in perf), and cleanup related interface to allow that to happen
4. Fix up the related tests

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.Core.Test/DefaultActionSelectorTests.cs [hunk 1]
```diff
@@ -860,7 +861,7 @@ private static RouteContext CreateRouteContext(string httpMethod)
             var routeData = new RouteData();
             routeData.Routers.Add(new Mock<IRouter>(MockBehavior.Strict).Object);
 
-            var httpContext = new Mock<HttpContext>(MockBehavior.Strict);
+            var httpContext = new Mock<HttpContext>(MockBehavior.Loose);
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/DefaultActionSelectorTests.cs]: This makes me nervous. Can you just set the properties that are being used in the test (assuming it's the service provider).

**Comment by rynowak** [comment]
:shipit: after looking at changing mock back to strict

File: test/Microsoft.AspNet.Mvc.Core.Test/DefaultActionSelectorTests.cs (hunk 1 above)
Comment (yishaigalatzer) [inline test/Microsoft.AspNet.Mvc.Core.Test/DefaultActionSelectorTests.cs]: It makes a big messy to the tests, I'm not sure how important this is, lets talk in person.

Comment (yishaigalatzer) [inline test/Microsoft.AspNet.Mvc.Core.Test/DefaultActionSelectorTests.cs]: We are going to make it strict, and pass services, as well as remove setting services on HttpContext from the test above

**Pull request closed without merging**